src/yachtzee.ts

```diff
@@ -0,0 +1,831 @@
+  // 1) Front to back of the schedule
+  // 2) Large spleen damagers to small spleen damagers
+  // This works because stench jellies are of size 1, so we can always pack efficiently using the greedy approach
+  remainingMenu.sort((left, right) => {
+    return right.spleen - left.spleen;
+  });
+
+  // Schedule jellies last so we definitely get spleen buffs first (e.g. synth and cologne)
+  for (const spleeners of [remainingMenu, jellies]) {
+    for (const entry of spleeners) {
+      let idx = 0;
+      let spleenUse = mySpleenUse();
+      while (
+        idx < dietSchedule.length &&
+        (dietSchedule[idx].spleen >= 0 || // We only insert if there's a cleanser immediately after where we want to insert
+          spleenUse + entry.spleen > spleenLimit() || // But don't insert if we will overshoot our spleen limit
+          (idx > 0 &&
+            dietSchedule[idx - 1].spleen < 0 &&
+            spleenUse + dietSchedule[idx].spleen >= 0)) // And cluster spleen cleansers (continue if the next cleanser can still clean our spleen)
+      ) {
+        spleenUse += dietSchedule[idx++].spleen ?? 0;
+      }
+      dietSchedule.splice(idx, 0, entry);
+    }
+  }
+
+  // Next, combine clustered entries where possible (this is purely for aesthetic reasons)
+  let idx = 0;
+  while (idx < dietSchedule.length - 1) {
+    if (dietSchedule[idx].name === dietSchedule[idx + 1].name) {
+      dietSchedule.splice(idx, 2, combineDietEntries(dietSchedule[idx], dietSchedule[idx + 1]));
+    } else idx++;
+  }
+
+  // Print diet schedule
+  print("Diet schedule:", "blue");
+  for (const entry of dietSchedule) print(`Use ${entry.quantity} ${entry.name}`, "blue");
+
+  // Finally, run a check to ensure everything is fine
+  let fullness = myFullness();
+  let drunkenness = myInebriety();
+  let spleenUse = mySpleenUse();
+  for (const entry of dietSchedule) {
+    fullness += entry.fullness;
+    drunkenness += entry.drunkenness;
+    spleenUse += entry.spleen;
+    if (fullness > fullnessLimit()) throw "Error in diet schedule: Overeating";
+    else if (drunkenness > inebrietyLimit()) throw "Error in diet schedule: Overdrinking";
+    else if (spleenUse > spleenLimit()) throw "Error in diet schedule: Overuse of spleen";
+  }
+
+  return dietSchedule;
+}
+
+export function yachtzeeChainDiet(simOnly?: boolean): boolean {
+  if (property.getBoolean("_garboYachtzeeChainDietPlanned")) return true;
+
+  // Plan for Yachtzee Chain
+  // 1) Fish Juice Box + Fishy Pipe for 30 turns of Fishy and Really Deep Breath (so we can ignore underwater gear)
+  // 2) PYEC to extend Fishy + Really Deep Breath if possible
+  // 3) Chew 30 (or 35 with PYEC) Stench Jellies (ensure that we have enough organ space)
+  // 4) Find meat and famwt buff that makes sense for a 2k base drop
+  // 5) Plant underwater friar's plant if possible
+
+  const havePYECCharge = have($item`Platinum Yendorian Express Card`) && !get(`expressCardUsed`);
```

Comment:
Most players will be getting a PYEC from their clan--is there a way for us to check that as well?

Reply:
We can probably try to pull 1 from the stash at the start if we have none on hand

Reply:
Unlike garbo's nominal behaviour which is not dependent on having stash access, our chaining diet heavily depends on whether we have access to a PYEC or not. I see 2 ways of implementing this.
1. We wrap `_yachtzeeChain()` in the scope of `withStash()` if we do not own a PYEC, but that requires holding on to the PYEC until we finish chaining (which honestly isn't that long), or;
2. We check if we have access to the PYEC prior to running `_yachtzeeChain()`, and set a global/mafia pref. This means that we only grab the PYEC when we need it. I'm not sure how I should parse `clan_stash.php` at this moment, as I don't know the exact details of how karma and stash restrictions interact with stash pulls. The simple assumption is that if I can pull a PYEC once, I can pull it infinitely many times per day (which is probably a fair assumption if the clan places enough trust in a member to not abscond with the PYEC in the first pull). Thus, we attempt to pull and immediately return the PYEC, and setting a _pref stating if we have access to a PYEC.

I've already pushed a commit for the 2nd implementation.